src/metal/window-proxy.ts

```diff
@@ -43,19 +47,35 @@ let W: WindowProxy = {
   getScrollLeft: nop,
   getHeight: nop,
   getWidth: nop,
+  rootsList: [],
   onWindowIsDirtyListeners: [],
   rAF: hasRAF ? window.requestAnimationFrame.bind(window) : (callback: Function) => { callback(); },
+  bindRootScrollEvent: (root: Element) => { root.addEventListener('scroll', () => W.callIsDirtyListeners(), false); },
+  callIsDirtyListeners() {
+    if (this.onWindowIsDirtyListeners.length <= 0) { return; }
+
+    this.onWindowIsDirtyListeners.forEach((obj: any) => {
+      let { fn, scope } = obj;
+      fn.call(scope);
+    });
+  },
   meta: {
     width: 0,
     height: 0,
     scrollTop: 0,
     scrollLeft: 0
   },
-  __destroy__() {
+  disconnectIsDirtyListener(id: string) {
+    this.onWindowIsDirtyListeners = this.onWindowIsDirtyListeners.filter((obj: any) => {
```

Comment:
if this becomes a map, this would become: `this.onWindowIsDirtyListeners.delete(id)`

Reply:
if  we do the above `VERSION` + `instance._lastVersion` stuff, this can be removed entirely.